sentry_sdk/_werkzeug.py

```diff
@@ -41,58 +41,89 @@
 
 
 #
-# `get_headers` comes from `werkzeug.datastructures.EnvironHeaders`
-# https://github.com/pallets/werkzeug/blob/0.14.1/werkzeug/datastructures.py#L1361
+# `get_headers` comes from `werkzeug.datastructures.headers.__iter__`
+# https://github.com/pallets/werkzeug/blob/3.1.3/src/werkzeug/datastructures/headers.py#L644
 #
 # We need this function because Django does not give us a "pure" http header
 # dict. So we might as well use it for all WSGI integrations.
 #
 def _get_headers(environ):
     # type: (Dict[str, str]) -> Iterator[Tuple[str, str]]
-    """
-    Returns only proper HTTP headers.
-    """
     for key, value in environ.items():
-        key = str(key)
-        if key.startswith("HTTP_") and key not in (
+        if key.startswith("HTTP_") and key not in {
             "HTTP_CONTENT_TYPE",
             "HTTP_CONTENT_LENGTH",
-        ):
+        }:
             yield key[5:].replace("_", "-").title(), value
-        elif key in ("CONTENT_TYPE", "CONTENT_LENGTH"):
+        elif key in {"CONTENT_TYPE", "CONTENT_LENGTH"} and value:
             yield key.replace("_", "-").title(), value
 
 
 #
 # `get_host` comes from `werkzeug.wsgi.get_host`
-# https://github.com/pallets/werkzeug/blob/1.0.1/src/werkzeug/wsgi.py#L145
+# https://github.com/pallets/werkzeug/blob/3.1.3/src/werkzeug/wsgi.py#L86
 #
 def get_host(environ, use_x_forwarded_for=False):
     # type: (Dict[str, str], bool) -> str
     """
     Return the host for the given WSGI environment.
     """
-    if use_x_forwarded_for and "HTTP_X_FORWARDED_HOST" in environ:
-        rv = environ["HTTP_X_FORWARDED_HOST"]
-        if environ["wsgi.url_scheme"] == "http" and rv.endswith(":80"):
-            rv = rv[:-3]
-        elif environ["wsgi.url_scheme"] == "https" and rv.endswith(":443"):
-            rv = rv[:-4]
-    elif environ.get("HTTP_HOST"):
-        rv = environ["HTTP_HOST"]
-        if environ["wsgi.url_scheme"] == "http" and rv.endswith(":80"):
-            rv = rv[:-3]
-        elif environ["wsgi.url_scheme"] == "https" and rv.endswith(":443"):
-            rv = rv[:-4]
-    elif environ.get("SERVER_NAME"):
-        rv = environ["SERVER_NAME"]
-        if (environ["wsgi.url_scheme"], environ["SERVER_PORT"]) not in (
-            ("https", "443"),
-            ("http", "80"),
-        ):
-            rv += ":" + environ["SERVER_PORT"]
-    else:
-        # In spite of the WSGI spec, SERVER_NAME might not be present.
-        rv = "unknown"
-
-    return rv
+    return _get_host(
+        environ["wsgi.url_scheme"],
+        (
+            environ.get("HTTP_HOST")
+            if not use_x_forwarded_for
+            else environ.get("HTTP_X_FORWARDED_HOST")
+        ),
```

Comment:
🎗️ The previous condition was: `if use_x_forwarded_for and "HTTP_X_FORWARDED_HOST" in environ:`

But if we have `use_x_forwarded_for` in True doesn't imply `HTTP_X_FORWARDED_HOST` will be present?